models/tags.js

```diff
@@ -0,0 +1,57 @@
+const db = require('../db');
+
+class tags {
+  static async createTag(entry_id,tag) {
+    // First have to find out if the tag already exist and if it does grab the tag id and make an insertion into journal_entries_tag table (the table takes an entry_id and a tag_id). If the tag doesn't exist it will first creat the tag and then do the function mentioned previously. 
+    const id = await db.query(`SELECT tags.name,tags.id FROM tags WHERE (tags.name = $1)`,[tag])
+     if (id.rows.length = 0){
+      await db.query(`INSERT INTO tags(name) VALUES($1)`,[tag])
+      await db.query(`SELECT tags.name,tags.id FROM tags WHERE (tags.name = $1)`,[tag])
+      db.query(`INSERT INTO journal_entries_tag(entry_id,tag_id)`,[entry_id,tagId])
+     }else{
+      let tagId = id.rows[0].id
+      db.query(`INSERT INTO journal_entries_tag(entry_id,tag_id)`,[entry_id,tagId])
+     }
+  }
+
+
+  static async getAllOfATag(tag){
+  // Is going to go into all of the entries if is_private is false and the tag is matching it will be rendered
+    let res = await db.query(`SELECT journal_entries.* FROM journal_entries
+    JOIN journal_entries_tag
+    ON (journal_entries_tag.entry_id = journal_entries.entry_id)
+    JOIN tags
+    ON (journal_entries_tag.tag_id = tags.id)
+    WHERE (tags.name = $1)
+    WHERE (journal_entries.is_Private = false)`,[tag])
+    return res.rows
```

Comment:
Sick SQL! Consider another name for method wasn't clear until reading code. Lint this page too, variables that aren't reassigned should be `const`. The parameters of your functions just become variables with values of the arguments so they don't need to be snaked_cased and can match the rest of your document being camelCased.